pkg/component/runtime/service_command.go

```diff
@@ -168,18 +175,14 @@ func (cr *cmdRetrier) Start(
 
 	backoffCtx := backoff.WithContext(expBackoff, retryCtx)
 
-	// Since we will be executing the command with infinite retries, we don't
-	// want to block.  So we execute the command with retries in its own
-	// goroutine.
-	go func() {
+	retryFunc := func() error {
```

Comment:
If you end up having to change the code anyway, Aleks pointed out that the contexts created here are only cancelled when `Stop` is called but not on the success path.

I tried a quick fix but one of the tests failed, and then I ran out of time to investigate.